packages/web3torrent/src/components/torrent-info/TorrentInfo.tsx

```diff
@@ -52,22 +63,26 @@ const TorrentInfo: React.FC<TorrentInfoProps> = ({
         <DownloadInfo torrent={torrent} />
       )}
 
-      <button
-        id="cancel-download-button"
-        type="button"
-        className="button cancel"
-        onClick={() => {
-          track('Torrent Cancelled', {
-            infoHash: torrent.infoHash,
-            magnetURI: torrent.magnetURI,
-            filename: torrent.name,
-            filesize: torrent.length
-          });
-          return web3TorrentClient.cancel(torrent.infoHash);
-        }}
-      >
-        Stop Torrenting
-      </button>
+      {!!torrent && !canWithdraw && (
+        <button
+          id="cancel-download-button"
+          type="button"
+          disabled={buttonDisabled}
+          className="button cancel"
+          onClick={() => {
+            track('Torrent Cancelled', {
+              infoHash: torrent.infoHash,
+              magnetURI: torrent.magnetURI,
+              filename: torrent.name,
+              filesize: torrent.length
+            });
+            setButtonDisabled(true);
+            return web3TorrentClient.cancel(torrent.infoHash);
+          }}
+        >
+          Stop {torrent && torrent.status === Status.Seeding ? 'Seeding' : 'Downloading'}
```

Comment:
Does this work after the leecher finishes the download but is still seeding?

Reply:
Fixed here 1efdbb2